docs/proposals/resize-when-creating-by-group.md

```diff
@@ -0,0 +1,127 @@
+
+## Summary
+
+Expand the PVC's initial capacity based on the largest capacity in specified PVCs.
+
+### Motivation
+
+One use case for pvc-autoresizer is automatically expanding a PV/PVC for multiple Pod/PV pairs, such as a MySQL cluster.
+In such a cluster, if a Pod/PV fails and a PV/PVC is rebuilt by the restore process, a PV is created based on the PVC template. Even if a PV of the same size as the others is actually needed, a PV of the template's capacity is created. As a result, the restore process stops due to insufficient capacity.
+
+To solve this problem, we are planning to add a new feature to create a PV/PVC with sufficient capacity when PVCs of the same group already exist.
+
+### Goals
+
+- Provide the ability to increase the PVC's initial capacity based on the largest capacity in specified PVCs.
+- The PVCs for which the initial capacity increase is enabled can be selected by users.
+- [MOCO](https://github.com/cybozu-go/moco) checks the PV capacity at the beginning of the restore process. It should be able to handle such cases.
+
+## Proposal
+
+### Rules for grouping & initial resize
+
+- If a PVC being created has a `resize.topolvm.io/pre-resize-group-by` annotation and the label specified in the annotation exists, existing PVCs with matching label values are considered to be in the same group.
+- If no PVCs are in the same group, or if the capacity of creating PVC is larger than any existing PVCs in the same group, the capacity of the original PVC is used as is.
+
+Examples are given below.
+
+```yaml
+### existing PVCs (excerpted)
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-1
+  labels:
+    label-foobar: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-foobar
+spec:
+  resources:
+    requests:
+      storage: 20Gi
+
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-2
+  labels:
+    label-foobar: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-foobar
+spec:
+  resources:
+    requests:
+      storage: 16Gi
+
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-y-1
+  labels:
+    label-foobar: group-y
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-foobar
+spec:
+  resources:
+    requests:
+      storage: 30Gi
+```
+
+When creating the following new PVC, `pvc-x-1` and `pvc-x-2` with `label-hoge: group-x` are considered to be in the same group, and `pvc-y-1` is not. Therefore, the PVC is created with **20Gi** based on `pvc-x-1`, which has the largest capacity in the group.
+
+```yaml
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-x-3
+  labels:
+    label-foobar: group-x
+  annotations:
+    resize.topolvm.io/pre-resize-group-by: label-foobar
+spec:
+  resources:
+    requests:
+      storage: 10Gi
+```
+
+When creating the following new PVC, `pvc-y-1` with `label-hoge: group-y` is in the same group. However, since the new PVC's size(50Gi) is larger than the existing one(30Gi), the PVC is created with **50Gi**.
```

Comment:
```suggestion
When creating the following new PVC, `pvc-y-1` with `label-foobar: group-y` is in the same group. However, since the new PVC's size(50Gi) is larger than the existing one(30Gi), the PVC is created with **50Gi**.
```